lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+
+          when 'osx'
+            # Install from latest CRAN binary build for OS X
+            sh.cmd "wget #{config[:cran]}/bin/macosx/R-latest.pkg " +
+                   '-O /tmp/R-latest.pkg'
+
+            sh.echo 'Installing OS X binary package for R'
+            sh.cmd 'sudo installer -pkg "/tmp/R-latest.pkg" -target /'
+            sh.rm '/tmp/R-latest.pkg'
+
+          else
+            sh.failure "Operating system not supported: #{config[:os]}"
+          end
+
+          if needs_bioc?
+            setup_bioc
+          end
+          if config[:latex]
+            setup_latex
+          end
+          if config[:pandoc]
+            setup_pandoc
+          end
+        end
+
+        def announce
+          super
+
+          sh.cmd 'Rscript -e \'sessionInfo()\''
+          sh.echo ''
+        end
+
+        def install
+          super
+
+          # Install any declared packages
+          apt_install config[:apt_packages]
+          r_binary_install config[:r_binary_packages]
+          r_install config[:r_packages]
+          bioc_install config[:bioc_packages]
+          r_github_install config[:r_github_packages]
+
+          # Install dependencies for the package we're testing.
+          install_deps
+        end
+
+        def script
+          # Build the package
+          sh.echo "Building with: R CMD build ${R_BUILD_ARGS}"
+          sh.cmd "R CMD build #{config[:r_build_args]} ."
+          tarball_script = [
+            'library("devtools")',
+            'pkg <- as.package(".")',
+            'cat(paste0(pkg$package, "_", pkg$version, ".tar.gz"))',
+          ].join('; ')
+          sh.export 'PKG_TARBALL', "$(Rscript -e '#{tarball_script}')"
+
+          # Test the package
+          sh.echo 'Testing with: R CMD check "${PKG_TARBALL}" ' +
+                  "#{config[:r_check_args]}"
+          sh.export '_R_CHECK_CRAN_INCOMING_', as_r_boolean(
+                      config[:r_check_cran_incoming])
+          if config[:r_check_cran_incoming]
+            sh.echo "(CRAN incoming checks are off)"
+          end
+          sh.cmd "R CMD check \"${PKG_TARBALL}\" #{config[:r_check_args]}"
+
+          # Turn warnings into errors, if requested.
+          if config[:warnings_are_errors]
+            export_rcheck_dir
+            sh.if 'grep -q -R "${RCHECK_DIR}/**/*.00check" "WARNING"' do
+              sh.failure "Found warnings, treating as errors (as requested)."
+            end
+          end
+          
+          # Check revdeps, if requested.
+          if config[:r_check_revdep]
+            sh.echo "Checking reverse dependencies"
+            revdep_script = [
+              'library("devtools");',
+              'res <- revdep_check();',
+              'if (length(res) > 0) {',
+              ' revdep_check_summary(res);',
+              ' revdep_check_save_logs(res);',
+              ' q(status = 1, save = "no");',
+              '}',
+            ].join(' ')
```

Comment:
IIRC, that string will ultimately become something like

    insert;      r code;      here;

as a value, right? it's not super important, but it seemed like it would read nicer in the generated script without the extra spaces ...